src/macos/aarch64/vcpu.rs

```diff
@@ -240,19 +240,30 @@ impl VirtualCPU for XhyveCpu {
 									);
 									copy_env(syscmdval, &self.parent_vm.mem);
 								}
-								Hypercall::FileClose(sysclose) => hypercall::close(sysclose),
-								Hypercall::FileLseek(syslseek) => hypercall::lseek(syslseek),
+								Hypercall::FileClose(sysclose) => hypercall::close(
+									sysclose,
+									&mut self.parent_vm.file_mapping.lock().unwrap(),
+								),
+								Hypercall::FileLseek(syslseek) => hypercall::lseek(
+									syslseek,
+									&mut self.parent_vm.file_mapping.lock().unwrap(),
+								),
```

Comment:
As explained in https://github.com/hermit-os/uhyve/issues/631, we should avoid panicking in hypercall handling if possible.

Reply:
Are you referring to the `unwrap()`, or is this the wrong line?

Reply:
nvm

Reply:
Which one of the following options would you prefer?

- Should I remove _all_ errors in hypercalls, like #631 insinuates, including in the event of a memory violation?
- Should I rely on `try_lock()` instead of `lock().unwrap()`, and return an error if `UhyveFileMap` cannot be accessed?
- Should I manually drop the mutex using `std::mem::drop(mutex)` during the error handling?

Reply:
My impression is, that all errors in these hypercalls can be handled in the kernel, or am I overlooking something?

Reply:
Ah, I thought the `unwrap` refers to the hypercall function, not the mutex. Yeah, you can panic on the mutex. If the mutex gets poisoned, we have quite some different problems.

Reply:
Happy that this was cleared up, thanks for clarifying. I'm marking this as resolved.